src/com/carolinarollergirls/scoreboard/core/impl/ClockImpl.java

```diff
@@ -363,18 +359,13 @@ public UpdateClockTimerTask() {
 
         public void addClock(ClockImpl c, boolean quickAdd) {
             synchronized (coreLock) {
-                if (c.isMasterClock()) {
-                    masterClock = c;
-                }
-                if (c.isSyncTime() && !quickAdd) {
+                if (c.isSyncTime() && !quickAdd && !clocks.isEmpty()) {
                     // This syncs all the clocks to change second at the same time
-                    // with respect to the master clock
-                    long nowMs = currentTime % 1000;
-                    if (masterClock != null) {
-                        nowMs = masterClock.getTime() % 1000;
-                        if (masterClock.isCountDirectionDown()) {
-                            nowMs = (1000 - nowMs) % 1000;
-                        }
+                    // with respect to the running clocks.
+                    ClockImpl masterClock = clocks.get(0);
+                    long nowMs = masterClock.getTime() % 1000;
```

Comment:
This is hiding a class variable.

Reply:
Ah, forgot to remove that.

Let me dig into that test. Things were test-order dependent previously, but this change should have fixed that in passing....

Reply:
I've run all the tests for ~2.5 hours, with no failures.